tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc

```diff
@@ -132,6 +133,10 @@ Status GrpcServer::Init() {
         return errors::InvalidArgument(
             "Could not parse port for local server from \"", iter->second,
             "\"");
+      } else if (!port::IsValidIpAddr(hostname_port[0])) {
+        return errors::InvalidArgument(
+            "Invalid ip address for local server from \"", iter->second,
+            "\"");
```

Comment:
I don't think this solves issue #7328. This is only checking the address for the local task (which we then overwrite with `0.0.0.0` or `localhost` later in this method). Since the process running this code will (or at least should) never attempt to connect to itself, I very much doubt that it's the source of the hanging.

It'd probably be more valuable to add validation for the addresses that get added to the `GrpcChannelSpec` (on L180). These are the addresses that will be used for contacting other workers, which is probably where your code was blocking.

The more important place for validation would be when you build the `GrpcChannelSpec`